api/dimensions_test.go

```diff
@@ -40,6 +72,158 @@ func TestGetDimensionsReturnsOk(t *testing.T) {
+			So(len(mockedDataStore.GetVersionCalls()), ShouldEqual, 1)
+			So(len(mockedDataStore.GetDimensionsCalls()), ShouldEqual, 1)
+		})
+	})
+
+	Convey("When valid limit above maximum and offset query parameters are provided", t, func() {
+		r := httptest.NewRequest("GET", "http://localhost:22000/datasets/123/editions/2017/versions/1/dimensions?offset=1&limit=7", nil)
+		w := httptest.NewRecorder()
+		mockedDataStore := &storetest.StorerMock{
+			GetVersionFunc: func(datasetID, edition, version, state string) (*models.Version, error) {
+				return mongoVersion, nil
+			},
+			GetDimensionsFunc: func(datasetID, versionID string) ([]bson.M, error) {
+				return mongoData, nil
+			},
+		}
+
+		datasetPermissions := getAuthorisationHandlerMock()
+		permissions := getAuthorisationHandlerMock()
+		api := GetAPIWithMocks(mockedDataStore, &mocks.DownloadsGeneratorMock{}, datasetPermissions, permissions)
+		api.Router.ServeHTTP(w, r)
+
+		Convey("then return editions information according to the offset and limit", func() {
+			expectedResponse := models.DatasetDimensionResults{
+				Items: []models.Dimension{
+					{
+						Links: models.DimensionLink{
+							Options: models.LinkObject{
+								HRef: "http://localhost:22000/datasets/version_dataset_id/editions//versions/version_ID/dimensions//options",
+							},
+							CodeList: models.LinkObject{},
+							Version: models.LinkObject{
+								HRef: "http://localhost:22000/datasets/version_dataset_id/editions//versions/version_ID",
+							},
+						},
+					},
+				},
+				Count:      1,
+				Offset:     1,
+				Limit:      7,
+				TotalCount: 2,
+			}
+
+			So(w.Code, ShouldEqual, http.StatusOK)
+			validateBody(w.Body.Bytes(), expectedResponse)
+		})
+
+		Convey("then the expected calls are performed against the database", func() {
+			So(datasetPermissions.Required.Calls, ShouldEqual, 1)
+			So(permissions.Required.Calls, ShouldEqual, 0)
+			So(len(mockedDataStore.GetVersionCalls()), ShouldEqual, 1)
+			So(len(mockedDataStore.GetDimensionsCalls()), ShouldEqual, 1)
+		})
+	})
+
+	Convey("When a negative limit and offset query parameters are provided", t, func() {
+		r := httptest.NewRequest("GET", "http://localhost:22000/datasets/123/editions/2017/versions/1/dimensions?offset=-1&limit=-7", nil)
+		w := httptest.NewRecorder()
+		mockedDataStore := &storetest.StorerMock{
+			GetVersionFunc: func(datasetID, edition, version, state string) (*models.Version, error) {
+				return mongoVersion, nil
+			},
+			GetDimensionsFunc: func(datasetID, versionID string) ([]bson.M, error) {
+				return []bson.M{
+					bson.M{"doc": bson.M{}},
+				}, nil
+			},
+		}
+
+		datasetPermissions := getAuthorisationHandlerMock()
+		permissions := getAuthorisationHandlerMock()
+		api := GetAPIWithMocks(mockedDataStore, &mocks.DownloadsGeneratorMock{}, datasetPermissions, permissions)
+		api.Router.ServeHTTP(w, r)
+
+		Convey("then return editions information with offset and limit equal to zero", func() {
```

Comment:
This seems incorrect functionality, surely the parameters are incorrect and we should inform the user they have made a mistake?